arangod/Transaction/Manager.cpp

```diff
@@ -472,10 +472,8 @@ Result Manager::createManagedTrx(TRI_vocbase_t& vocbase, TransactionId tid,
   // start the transaction
   transaction::Hints hints;
   hints.set(transaction::Hints::Hint::GLOBAL_MANAGED);
-  if (options.isFollowerTransaction || tid.isFollowerTransactionId()) {
-    hints.set(transaction::Hints::Hint::IS_FOLLOWER_TRX);
-  }
   if (isFollowerTransactionOnDBServer) {
+    hints.set(transaction::Hints::Hint::IS_FOLLOWER_TRX);
```

Comment:
In principle a good change, I fear this might break the unit tests, since there are cases in there which do not set the server role to dbserver, as far as I remember. Let's see. If the unit tests are OK, then I am fine with this.

Reply:
This change was intended to be just a code simplification.
The 2 identical `if` statements are merged here, which shouldn't change anything.
Or am I missing something?

Reply:
I fear `isFollowerTransactionOnDBServer` checks if we are on a DBServer. In the unit tests (and Mocking) this is mostly `false`, so the if will no longer be triggered.
I might be wrong, though.

Reply:
You were right, the one unit test failed. I fixed it by setting the server role to DB server for this particular test.